src/components/content/monitor/Monitor.tsx

```diff
@@ -0,0 +1,326 @@
+import { fetchMonitorMetricDataTimeInterval, monitorMetricQueueSize } from '../../utils/constants';
+
+export interface MetricProps {
+    id: string;
+
+    name: string;
+
+    value: number;
+
+    vmName: string;
+}
+
+function Monitor(): JSX.Element {
+    const [form] = Form.useForm();
+    const [serviceId, setServiceId] = useState<string>('');
+    const [monitorChartItems, setMonitorChartItems] = useState<Tab[]>([]);
+    const [deployedServiceList, setDeployedServiceList] = useState<ServiceVo[]>([]);
+    const [fetchMonitorMetricDataTimer, setFetchMonitorMetricDataTimer] = useState<NodeJS.Timeout>();
+    const monitorMetricsQueue: Queue<Metric[]> = arrayQueue<Metric[]>(monitorMetricQueueSize);
+    const [tip, setTip] = useState<JSX.Element | undefined>(undefined);
+    const [queryResult, setQueryResult] = useState<boolean>(false);
+    const [serviceNameList, setServiceNameList] = useState<{ value: string; label: string }[]>([
+        { value: '', label: '' },
+    ]);
+    const [customerServiceNameList, setCustomerServiceNameList] = useState<{ value: string; label: string }[]>([
+        { value: '', label: '' },
+    ]);
+
+    function Tip(type: 'error' | 'success', msg: string) {
+        setTip(
+            <div className={'submit-alert-tip'}>
+                {' '}
+                <Alert message='Operating System Monitor:' description={msg} showIcon type={type} />{' '}
+            </div>
+        );
+    }
+
+    function TipClear() {
+        setTip(undefined);
+    }
+
+    useEffect(() => {
+        void ServiceService.listDeployedServices()
+            .then((rsp: ServiceVo[]) => {
+                const serviceNameList: { value: string; label: string }[] = [];
+                const cusServiceNameList: { value: string; label: string }[] = [];
+                if (rsp.length > 0) {
+                    rsp.forEach((serviceVo: ServiceVo) => {
+                        const cusServiceName: { value: string; label: string } = {
+                            value: serviceVo.customerServiceName ?? '',
+                            label: serviceVo.customerServiceName ?? '',
+                        };
+                        cusServiceNameList.push(cusServiceName);
+                        const serviceName: { value: string; label: string } = {
+                            value: serviceVo.name,
+                            label: serviceVo.name,
+                        };
+
+                        serviceNameList.push(serviceName);
+                    });
+                    setDeployedServiceList(rsp);
+                    setServiceNameList(serviceNameList);
+                    setCustomerServiceNameList(cusServiceNameList);
+                }
+            })
+            .catch((error: Error) => {
+                setDeployedServiceList([]);
+                setServiceNameList([]);
+                setCustomerServiceNameList([]);
+                Tip('error', 'fetch all deployed services error.');
```

Comment:
Error message can be -> Error while fetching all deployed services.

Reply:
fixed.